test/test.js

```diff
@@ -564,6 +565,18 @@ function tests(dbName, dbType) {
         ids.should.deep.equal(['2']);
       });
     });
+    it('allows searching from an array of nested objects', function () {
+      return db.bulkDocs({docs: docs9}).then(function () {
+        var opts = {
+          fields: ['nested.array.aField'],
```

Comment:
I was a bit confused at first that this was conflating deep fields and array search, but I understand now that both features are supported independently. :)